conftest.py

```
@@ -58,12 +71,44 @@ def reset_environment():
     os.environ.setdefault("ALLOW_DEV_API_KEY", "true")
     os.environ.setdefault("PYTHONPATH", ".:core:app:tests")
 
+    # Override API key validation for all tests
+    try:
+        import app as app_module
+        from app import app as fastapi_app
+
+        # Simple pass-through that accepts any non-empty API key
+        def mock_get_api_key(api_key: str = None):
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)